### DISCUSSION TITLE
Custom rules not showing errors/warning in Android studio IDE

### DISCUSSION THREAD

**Discussion by The-Pascal** [detekt/detekt#7595] [General]
I have added multiple custom rules in my project, and added them to my detekt.yml.
**My custom rules are working fine when running with CLI, but custom rules do not shows any error/warning while I am coding in my IDE.**

However, if I create JAR of my module and add it to plugin explicitly then custom rules starts showing errors in my IDE.

`IDE` - Android Studio Koala | 2024.1.1 Patch 1
`Detekt Plugin` - 2.4.1

Is this behaviour expected?
I want that if I have custom rules in the same project, then I should not have to create JAR and add it explicitly.

**Comment by cortinico** [comment]
You should report this here: https://github.com/detekt/detekt-intellij-plugin

**Comment by renatojobal** [comment]
If you mean that the IDE is not taking in account your custom rules, then you should add the rules files to the detekt plugin at Settings > Tools > detek > Files :

![image](https://github.com/user-attachments/assets/3cfd266f-78a1-428d-bd1f-11699c34622b)
